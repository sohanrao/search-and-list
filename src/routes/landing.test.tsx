import { RouterProvider, createMemoryRouter } from "react-router-dom";
import { describe, it, expect, beforeEach } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Landing from "./landing";
import Index from ".";

const routes = [
  {
    path: "/",
    element: <Landing />,
    children: [
      {
        children: [
          { index: true, element: <div>Index route</div> },
          {
            path: ":type/:value",
            element: <div>Route</div>,
          },
        ],
      },
    ],
  },
];

const router = createMemoryRouter(routes, {
  initialEntries: ["/"],
});

describe("Landing Page", () => {
  it("renders the Landing Page component", async () => {
    const { container } = render(<RouterProvider router={router} />);
    expect(container).toMatchSnapshot();
  });
  describe("Landing Page content", () => {
    beforeEach(() => {
      render(<RouterProvider router={router} />);
    });
    it("should find only one instance of app name and search input", () => {
      const names = screen.queryAllByText("All You Can Read");
      const inputs = screen.queryAllByPlaceholderText("Search by title");
      expect(names.length).toBe(1);
      expect(inputs.length).toBe(1);
    });
    it("should set tag to active when clicked", async () => {
      await userEvent.click(screen.getByText("self-help"));
      expect(screen.getByText("self-help").classList.contains("active")).toBe(
        true,
      );
    });
  });
});
