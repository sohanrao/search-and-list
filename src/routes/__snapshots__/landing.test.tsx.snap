// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Landing Page > renders the Landing Page component 1`] = `
<div>
  <div
    class="app__header"
  >
    <div>
      <a
        class="app__header--logo"
        href="/"
      >
        All You Can Read
      </a>
    </div>
    <div
      class="app__search"
    >
      <input
        name="search"
        placeholder="Search by title"
        type="search"
      />
      <a
        class="app__search--button"
        href="/search?q="
      >
        Search
      </a>
    </div>
    <div
      class="app__search--tags"
    >
      <span>
        Popular categories:
      </span>
      <a
        class=""
        href="/genre/fiction"
      >
        fiction
      </a>
      <a
        class=""
        href="/genre/non-fiction"
      >
        non-fiction
      </a>
      <a
        class=""
        href="/genre/self-help"
      >
        self-help
      </a>
      <a
        class=""
        href="/genre/biography"
      >
        biography
      </a>
      <a
        class=""
        href="/genre/education"
      >
        education
      </a>
      <a
        class=""
        href="/genre/food"
      >
        food
      </a>
      <a
        class=""
        href="/genre/music"
      >
        music
      </a>
      <a
        class=""
        href="/genre/movies"
      >
        movies
      </a>
      <a
        class=""
        href="/genre/crime"
      >
        crime
      </a>
    </div>
  </div>
  <div
    class=""
    id="detail"
  >
    <div>
      Index route
    </div>
  </div>
</div>
`;
